### PR TITLE
scaffolding for batch announcer

### PR DESCRIPTION
**Pull request by saraswatpuneet**
Part of #12 

## Details
* Basic scaffolding for batch annoucer as it will over lap in parts with batch builder processor
* Added a module and respective place holder files/functions for a batch announcer
* It will listen to `batchAnnouncementQueue` followed by announcement on IPFS -> queue  to `publishQueue`

### CODE REVIEW

File: apps/worker/src/batch_announcer/batch.announcer.module.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            password: password || undefined,
+            db: pathname?.length > 1 ? Number(pathname.slice(1)) : undefined,
+          },
+        };
+      },
+      inject: [ConfigService],
+    }),
+    BullModule.registerQueue(
+      {
+        name: 'publishQueue',
```
Comment (aramikm) [inline apps/worker/src/batch_announcer/batch.announcer.module.ts:53]: nit: please use defined constants in `QueueConstants`

File: apps/worker/src/batch_announcer/batch.announcer.module.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          attempts: 1,
+          backoff: {
+            type: 'exponential',
+          },
+          removeOnComplete: true,
+          removeOnFail: false,
+        },
+      },
+      {
+        name: 'batchAnnouncerQueue',
```
Comment (aramikm) [inline apps/worker/src/batch_announcer/batch.announcer.module.ts:64]: same here

File: apps/worker/src/batch_announcer/batch.announcer.service.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import Redis from 'ioredis';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { ConfigService } from '../../../api/src/config/config.service';
+import { IPFSAnnouncer } from './ipfs.announcer';
+import { CAPACITY_EPOCH_TIMEOUT_NAME } from '../../../../libs/common/src/constants';
+import { IBatchAnnouncerJobData } from '../interfaces/batch-announcer.job.interface';
+
+@Injectable()
+@Processor('batchAnnouncerQueue', {
```
Comment (aramikm) [inline apps/worker/src/batch_announcer/batch.announcer.service.ts:14]: same

File: apps/worker/src/batch_announcer/batch.announcer.service.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  concurrency: 2,
+})
+export class BatchAnnouncementService extends WorkerHost implements OnApplicationBootstrap, OnModuleDestroy {
+  private logger: Logger;
+
+  private capacityExhausted = false;
+
+  constructor(
+    @InjectRedis() private cacheManager: Redis,
+    @InjectQueue('publishQueue') private publishQueue: Queue,
```
Comment (aramikm) [inline apps/worker/src/batch_announcer/batch.announcer.service.ts:24]: same

File: apps/worker/src/interfaces/batch-announcer.job.interface.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export interface IBatchAnnouncerJobData {
```
Comment (aramikm) [inline apps/worker/src/interfaces/batch-announcer.job.interface.ts:1]: shouldn't this have a list of all the announcements in the batch?

**Review by aramikm — APPROVED** [review]
Overall looks good, just some nits!

File: apps/worker/src/batch_announcer/batch.announcer.service.ts (hunk 4 above)
Comment (saraswatpuneet) [reply apps/worker/src/batch_announcer/batch.announcer.service.ts:24]: fixed in this https://github.com/AmplicaLabs/content-publishing-service/pull/26

File: apps/worker/src/interfaces/batch-announcer.job.interface.ts (hunk 5 above)
Comment (saraswatpuneet) [reply apps/worker/src/interfaces/batch-announcer.job.interface.ts:1]: I intend to do that when logic is being put in place

File: apps/worker/src/batch_announcer/batch.announcer.service.ts (hunk 3 above)
Comment (saraswatpuneet) [reply apps/worker/src/batch_announcer/batch.announcer.service.ts:14]: fixed in this https://github.com/AmplicaLabs/content-publishing-service/pull/26

File: apps/worker/src/batch_announcer/batch.announcer.module.ts (hunk 2 above)
Comment (saraswatpuneet) [reply apps/worker/src/batch_announcer/batch.announcer.module.ts:64]: fixed in this https://github.com/AmplicaLabs/content-publishing-service/pull/26

**Pull request merged**